java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrow.java

```diff
@@ -110,13 +112,36 @@ public static VectorSchemaRoot sqlToArrow(
       String query,
       BaseAllocator allocator,
       Calendar calendar) throws SQLException, IOException {
+
     Preconditions.checkNotNull(connection, "JDBC connection object can not be null");
     Preconditions.checkArgument(query != null && query.length() > 0, "SQL query can not be null or empty");
     Preconditions.checkNotNull(allocator, "Memory allocator object can not be null");
     Preconditions.checkNotNull(calendar, "Calendar object can not be null");
 
+    return sqlToArrow(connection, query, new JdbcToArrowConfig(allocator, calendar));
+  }
+
+  /**
+   * For the given SQL query, execute and fetch the data from Relational DB and convert it to Arrow objects.
+   *
+   * @param connection Database connection to be used. This method will not close the passed connection object.
```

Comment:
it might be a simpler contract to pass through the ResultSet object instead here?

Reply:
That version exists on line 224.

This version can certainly be removed, but I was trying to preserve the pair of contracts the original author supplied.  The two contracts were:

1. <Database + Query> + [Allocator | Calendar | Both | Neither]
2. <ResultSet> + [Allocator | Calendar | Both | Neither]

Since I'm adding a new contract of <ResultSet> + <Config>, it made sense to me to also add <Database + Query> + <Config>.